bin/prover-client/src/handlers/cl_stf.rs

```diff
@@ -0,0 +1,130 @@
+use std::sync::Arc;
+
+use jsonrpsee::{core::client::ClientT, http_client::HttpClient, rpc_params};
+use strata_db::traits::ProofDatabase;
+use strata_primitives::{
+    buf::Buf32,
+    proof::{ProofContext, ProofKey},
+};
+use strata_proofimpl_cl_stf::prover::{ClStfInput, ClStfProver};
+use strata_rocksdb::prover::db::ProofDb;
+use strata_rpc_types::RpcBlockHeader;
+use strata_state::id::L2BlockId;
+use tokio::sync::Mutex;
+use tracing::error;
+
+use super::{evm_ee::EvmEeHandler, utils::get_pm_rollup_params, ProvingOp};
+use crate::{errors::ProvingTaskError, hosts, task::TaskTracker};
+
+/// Operations required for CL block proving tasks.
+#[derive(Debug, Clone)]
+pub struct ClStfHandler {
+    cl_client: HttpClient,
+    evm_ee_handler: Arc<EvmEeHandler>,
+}
+
+impl ClStfHandler {
+    /// Creates a new CL operations instance.
+    pub fn new(cl_client: HttpClient, evm_ee_handler: Arc<EvmEeHandler>) -> Self {
+        Self {
+            cl_client,
+            evm_ee_handler,
+        }
+    }
+
+    pub async fn get_id(&self, block_num: u64) -> Result<L2BlockId, ProvingTaskError> {
+        let l2_headers: Option<Vec<RpcBlockHeader>> = self
+            .cl_client
+            .request("strata_getHeadersAtIdx", rpc_params![block_num])
+            .await
+            .inspect_err(|_| error!(%block_num, "Failed to fetch l2_headers"))
+            .map_err(|e| ProvingTaskError::RpcError(e.to_string()))?;
+
+        let cl_stf_id_buf: Buf32 = l2_headers
+            .expect("invalid height")
+            .first()
+            .expect("at least one l2 blockid")
+            .block_id
+            .into();
+        Ok(cl_stf_id_buf.into())
+    }
+
+    pub async fn get_slot(&self, id: L2BlockId) -> Result<u64, ProvingTaskError> {
+        let header: RpcBlockHeader = self
+            .cl_client
+            .request("strata_getHeaderById", rpc_params![id])
+            .await
+            .map_err(|e| ProvingTaskError::RpcError(e.to_string()))?;
+        Ok(header.block_idx)
```

Comment:
Should be using the RPC API via `jsonrpsee` instead of manually implementing a JSON-RPC client again.